rpcclient/infrastructure.go

```diff
@@ -159,6 +148,13 @@ type Client struct {
 	disconnect      chan struct{}
 	shutdown        chan struct{}
 	wg              sync.WaitGroup
+
+	// rrbalancer holds instance of RoundRobinBalancer. This implements
+	// the logic to round-robin between available connections.
+	// Whenever we need a connection, RoundRobinBalancer.NextConn() should be used to
+	// retrive a websocket connection or a HostAddress in case of HttpPostMode.
+	// It relies on the HostAddresses present in ConnConfig as the source for round-robin.
+	rrbalancer *RoundRobinBalancer
```

Comment:
I would suggest just calling this `balancer` so that it can potentially provide other load balancing methods in the future.